pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -571,6 +571,17 @@ func (ctrl *PersistentVolumeController) syncVolume(volume *v1.PersistentVolume)
 			}
 			return nil
 		} else if claim.Spec.VolumeName == "" {
+			if isMisMatch, err := checkVolumeModeMisMatches(&claim.Spec, &volume.Spec); err != nil || isMisMatch {
+				// Binding for the dynamically provisioned volume won't be called in syncUnboundClaim,
+				// because findBestMatchForClaim won't return the volume due to volumeMode mismatch.
+				volumeMsg := fmt.Sprintf("Dynamically provisioned volume[%s] has incompatible volumeMode. This provisioner may not support the requested volumeMode.", volume.Name)
+				ctrl.eventRecorder.Event(volume, v1.EventTypeWarning, events.VolumeMismatch, volumeMsg)
+				claimMsg := fmt.Sprintf("Dynamically provisioned PersistentVolume[%s] has an incompatible volumeMode to PersistentVolumeClaim[%s]. It won't be bound to this claim. Please create proper volume manually and delete this volume manually.", volume.Name, claim.Name)
```

Comment:
I would use almost the same message for both PV and PVC.  Maybe the only difference is that PV message gives the PVC name, and the PVC message gives the PV name.